lib/kv/manager/src/layout.rs

```diff
@@ -0,0 +1,142 @@
+// SPDX-FileCopyrightText: Copyright (c) 2024-2025 NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//! Block layout management.
+//!
+//! This module provides traits and implementations for managing how blocks
+//! are arranged in storage, including both contiguous and non-contiguous layouts.
+
+pub mod contiguous;
+
+use crate::dtype::DType;
+
+use derive_builder::Builder;
+use thiserror::Error;
+use tracing::instrument;
+use validator::Validate;
+
+/// Errors that can occur during layout operations
+#[derive(Debug, Error)]
+#[allow(missing_docs)]
+pub enum LayoutError {
```

Comment:
general: not clear to me what is a configuration vs layout vs layout type - but not showstopper